### PR TITLE
Data.Generic.Rep.Bounded

### PR DESCRIPTION
**Pull request by matthewleon**
Generic implementations of Prelude.Bounded class's top and bottom.

### CODE REVIEW

File: src/Data/Generic/Rep/Bounded.purs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+instance genericBottomNoArguments :: GenericBottom NoArguments where
+  genericBottom' = NoArguments
+
+instance genericBottomSum :: GenericBottom a => GenericBottom (Sum a b) where
+  genericBottom' = Inl genericBottom'
+
+instance genericBottomConstructor :: GenericBottom a => GenericBottom (Constructor name a) where
+  genericBottom' = Constructor genericBottom'
+
+instance genericBottomProduct :: (GenericBottom a, GenericBottom b) => GenericBottom (Product a b) where
```
Comment (paf31) [inline src/Data/Generic/Rep/Bounded.purs]: Do we want to support `Bounded` for product types? I think I would rather just support `NoArguments`.

Comment (matthewleon) [inline src/Data/Generic/Rep/Bounded.purs]: I have no personal objections to going the more conservative route. It's certainly more consistent with the GHC behavior. Should I modify the PR?

Comment (paf31) [inline src/Data/Generic/Rep/Bounded.purs]: Yes please. Let's keep things simple for now, and then when we add `Enum` deriving we'll have fewer questions 😄

**Comment by matthewleon** [comment]
@paf31 I've updated the branch to only support and test sum types with argument-free constructors.

**Comment by paf31** [comment]
Thanks!

**Pull request merged**